question16/main.js

```diff
@@ -83,21 +82,57 @@ const createTitles = ({ articles }) => {
     const titleList = document.createElement('li');
     const articleLink = document.createElement('a');
     articleLink.href = "#";
-    articleLink.textContent = articleTitle; 
+    articleLink.textContent = articleTitle;
+
+    // Display new icon
+    const elapsedTime = getElapsedDays(articles[i]);
+    if (elapsedTime <= 3) {
+      const newIcon = createElementWithClassName("span", "new_icon"); 
+      newIcon.textContent = 'new';
+      articleLink.appendChild(newIcon);      
+    }
+
+    //add comment icon and number
+    const commentsLength = articles[i].comments.length; 
+    if (commentsLength > 0) {
+      articleLink.appendChild(createCommentIcon(articles[i]));
+    }
+    
     fragmentTitles.appendChild(titleList).appendChild(articleLink);
   }
   return fragmentTitles;
 }
 
-//get img data 
-const createImgElements = (data) => {
+//Get img data 
+const createImgElements = (newsData) => {
   const imgWrapper = createElementWithClassName("div", "imgWrapper");
   const imgTag = document.createElement('img');
-  imgTag.src = data.img;
+  imgTag.src = newsData.img;
   imgWrapper.appendChild(imgTag);
   return imgWrapper;
 }
 
+// Display comment icons and numbers
+const createCommentIcon = (articlesData) => {
+  const commentLength = articlesData.comments.length;
+  const commentIconSpan = createElementWithClassName("span", "comment_icon");
+  const commentNum = createElementWithClassName("span", "comment_icon_num");
+  const commentIcon = createElementWithClassName("img", "comment_icon_img");
+  commentNum.textContent = commentLength;
+  commentIcon.src = './img/comment_icon.png';
+  commentIconSpan.appendChild(commentIcon);
+  commentIconSpan.appendChild(commentNum);
+  return commentIconSpan; 
+}
+
+// Get the number of days elapsed
+const getElapsedDays = (articlesData) => {
+const postedDate = new Date(articlesData.date);
+const today = new Date();
+const daysElapsed = (today - postedDate) / 86400000;
```

Comment:
`86400000` is ...?

Reply:
86400000ms ➡︎ 24hｘ60mｘ60sｘ1,000ms
It calculates milliseconds into days.
I referred to the following article. Also, I forgot to round up non-divisible numbers, so I added it(Math.floor). I just fixed here as following.
https://javascript.programmer-reference.com/js-date-term/

```js 
const getElapsedDays = (articlesData) => {
const postedDate = new Date(articlesData);
const today = new Date();
const elapsedDays = (today - postedDate) / 86400000;
return Math.floor(elapsedDays);
}
```

Latest commit : https://github.com/csakyo/homework/pull/30/commits/ccad1949a6adfc7ecda8ae44fcab7c62dce7d1a7